lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,309 @@
+                         enrollBtnInitial = this.$('.enroll-btn-initial');
+
+                     // Disable the button if the user is already enrolled in that session.
+                     if (currentSessionId === newSessionId) {
+                         enrollBtnInitial.addClass('disabled');
+                         enrollBtnInitial.popover('dispose');
+                         return;
+                     }
+                     enrollBtnInitial.removeClass('disabled');
+
+                     // Update button text specifying if the user is initially enrolling, changing or leaving a session.
+                     if (newSessionId) {
+                         enrollText = currentSessionId ? gettext('Change Session') : gettext('Select Session');
+                     } else {
+                         enrollText = gettext('Leave Current Session');
+                     }
+                     this.$('.enroll-btn-initial').text(enrollText);
+
+                     // Update the button popover text to enable two step authentication.
+                     if (newSessionId) {
+                         confirmationHTML = currentSessionId ?
+                             gettext('Are you sure that you would like to switch session?') + '<br><br>' +
+                             gettext(' Please know that by choosing to switch session you will lose any progress you' +
+                             'have made in this session.') :
+                             gettext('Are you sure that you would like to select this session?');
+                     } else {
+                         confirmationHTML = gettext('Are you sure that you would like to leave this session?') +
+                             '<br><br>' + gettext('Please know that by leaving you will lose any progress you had' +
+                             ' made in this session.');
+                     }
+
+                     // Remove existing popover and re-initialize
+                     enrollBtnInitial.popover('dispose');
+                     enrollBtnInitial.popover({
+                         placement: 'bottom',
+                         container: this.$el,
+                         html: true,
+                         trigger: 'click',
+                         content: '<div class="verification-modal" role="dialog"' +
+                             'aria-labelledby="enrollment-verification-title">' +
+                             '<p id="enrollment-verification-title">' + confirmationHTML + '</p>' +
+                             '<div class="action-items">' +
+                             '<button type="button" class="popover-dismiss final-confirmation-btn">' +
+                             gettext('No') + '</button>' +
+                             '<button type="button" class="enroll-btn final-confirmation-btn">' +
+                             gettext('Yes') + '</button></div></div>'
+                     });
```

Comment:
It might be easier to move all of this HTML content generation for the popover into its own underscore template?

Reply:
Moved